js/coinbasepro.js

```diff
@@ -429,55 +431,95 @@ module.exports = class coinbasepro extends Exchange {
+            timestamp = this.milliseconds ();
+            last = ticker;
+            info = { 'symbol': ticker };
+        } else {
+            timestamp = this.parse8601 (this.safeValue (ticker, 'time'));
+            bid = this.safeNumber (ticker, 'bid');
+            ask = this.safeNumber (ticker, 'ask');
+            high = this.safeNumber (ticker, 'high');
+            low = this.safeNumber (ticker, 'low');
+            open = this.safeNumber (ticker, 'open');
+            last = this.safeNumber2 (ticker, 'price', 'last');
+            volume = this.safeNumber (ticker, 'volume');
+            info = ticker;
+        }
+        return this.safeTicker ({
             'symbol': symbol,
             'timestamp': timestamp,
             'datetime': this.iso8601 (timestamp),
-            'high': this.safeNumber (ticker, 'high'),
-            'low': this.safeNumber (ticker, 'low'),
+            'high': high,
+            'low': low,
             'bid': bid,
             'bidVolume': undefined,
             'ask': ask,
             'askVolume': undefined,
             'vwap': undefined,
-            'open': this.safeNumber (ticker, 'open'),
+            'open': open,
             'close': last,
             'last': last,
             'previousClose': undefined,
             'change': undefined,
             'percentage': undefined,
             'average': undefined,
-            'baseVolume': this.safeNumber (ticker, 'volume'),
+            'baseVolume': volume,
             'quoteVolume': undefined,
-            'info': ticker,
-        };
+            'info': info,
+        });
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {};
+        // only publicGetSparkLines available atm
+        const method = this.safeString (this.options, 'fetchTickersMethod', 'publicGetProductsSparkLines');
+        const response = await this[method] (this.extend (request, params));
+        // array order : t-l-h-o-c
+        //  {
+        //      YYY-USD: [
+        //          [
+        //              1639472400,
+        //              4.26,
+        //              4.38,
+        //              4.35,
+        //              4.27
+        //          ],
+        //          [
+        //              1639468800,
+        //              4.31,
+        //              4.45,
+        //              4.35,
+        //              4.35
+        //          ]
+        //      ZZZ-USD : [
+        //          [
+        //              1639472400,
+        //              ...
+        const result = {};
+        const marketIds = Object.keys (response);
+        const delimiter = '-';
+        for (let i = 0; i < marketIds.length; i++) {
+            const marketId = marketIds[i];
+            const entry = this.safeValue (response, marketId, []);
+            const lastRow = this.safeValue (entry, 0, []);
+            const lastPrice = this.safeValue (lastRow, 4);
```

Comment:
Fetching tickers with klines is wrong, cause OHLCV candles are aligned to timestamp boundaries, but the ticker is a floating window of 24h. I think we should not emulate it in this case.